### PR TITLE
Add brief documentation on ic-cdk-optimizer usage

### PR DESCRIPTION
**Pull request by ililic**

### CODE REVIEW

File: src/ic-cdk-optimizer/README.md [hunk 1]
```diff
@@ -1,4 +1,63 @@
...
+# Installation
+To install the ic-cdk-optimizer with cargo run:
+
+`cargo install ic-cdk-optimizer --root target`
+
+# Usage
+ic-cdk-optimizer will take your existing `.wasm` files and strip unused data segments. So for a canister named [rust_hello](https://sdk.dfinity.org/docs/rust-guide/rust-quickstart.html) the following command produces an optimized `rust_hello-opt.wasm`:
+
+```
+./target/bin/ic-cdk-optimizer ./target/wasm32-unknown-unknown/debug/rust_hello.wasm -o ./target/wasm32-unknown-unknown/release/rust_hello-opt.wasm
```
Comment (hansl) [inline src/ic-cdk-optimizer/README.md:13]: With the dfx.json below you need to rerun the optimizer yourself as a separate flow. It's better to encapsulate both cargo build and ic-cdk-optimizer in a shell script.

**Review by hansl — APPROVED** [review]
LGTM if you make that change above.

File: src/ic-cdk-optimizer/README.md (hunk 1 above)
Comment (shanling) [reply src/ic-cdk-optimizer/README.md:13]: What I've done as of late is use the `cargo-make` plugin with a separate `Makefile.toml`

```
//Makefile.toml
[tasks.build_and_opt]
script = [
    """
    cargo build --target wasm32-unknown-unknown --package rust_hello &&
    ic-cdk-optimizer ~/path/to/project/target/wasm32-unknown-unknown/debug/rust_hello.wasm -o ~/path/to/project/target/wasm32-unknown-unknown/debug/rust_hello_opt.wasm
    """
]
```

in dfx.json
```
  "canisters": {
    "rust_hello": {
      "build": "cargo make build_and_opt",
      "candid" : "src/rust_hello/rust_hello.did",
      "wasm": "target/wasm32-unknown-unknown/debug/rust_hello_opt.wasm",
      "type" : "custom"
    }, ...
```
which is working quite well for me. might be a better way to do it, still quite new to Rust.

Comment (magj2006) [reply src/ic-cdk-optimizer/README.md:13]: Hi, I tried it following your way, but still got an error as below.

[cargo-make][1] ERROR - Task build_and_opt not found
[make.tar.gz](https://github.com/dfinity/cdk-rs/files/7631352/make.tar.gz)

[cargo-make][1] WARN - Build Failed.
[cargo-make] ERROR - Error while running duckscript: Source: Unknown Line: 4 - Error while executing command, exit code: 1
[cargo-make] WARN - Build Failed.

Comment (magj2006) [reply src/ic-cdk-optimizer/README.md:13]: > 

already solved.

**Pull request closed without merging**